financial-connections/src/main/java/com/stripe/android/financialconnections/image/StripeImage.kt

```diff
@@ -0,0 +1,113 @@
+package com.stripe.android.financialconnections.image
+
+import android.graphics.Bitmap
+import android.util.Log
+import android.util.LruCache
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.BoxWithConstraintsScope
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.unit.Dp.Companion.Infinity
+import androidx.compose.ui.unit.IntSize.Companion.Zero
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+@Composable
+internal fun StripeImage(
+    url: String,
+    placeHolder: Painter,
+    memoryCache: LruCache<String, Bitmap> = imageMemoryCache,
+    contentDescription: String? = null,
+    modifier: Modifier = Modifier
+) {
+    BoxWithConstraints(modifier) {
+        val (width, height) = calculateBoxSize()
+        val painter: MutableState<Painter> =
+            remember { mutableStateOf(BitmapPainter(ImageBitmap(width, height))) }
+        var loadImageJob: Job?
+        val scope = rememberCoroutineScope()
+
+        DisposableEffect(url) {
+            val stripeImageLoader = StripeImageLoader()
+            loadImageJob = scope.launch(Dispatchers.Default) {
+                painter.value = memoryCache.get(url)
```

Comment:
could we instead download the file to `context.cacheDir`?

Reply:
For Financial Connections use case, I was planning on just supporting a small in memory cache (we'll load about a dozen small icons in average) for simplicity and to avoid having to deal with File access related APIs.

Happy to iterate afterwards and add several cache layers as [Coil](https://coil-kt.github.io/coil/image_loaders/#caching) does. @ccen-stripe what do you think?

Reply:
Update - added disk cache and coordinated it with memory cache:  https://github.com/stripe/stripe-android/pull/5307/files#diff-c5d54219012c36b7353bc528722b5d4ce517254d1b7134e62386e43dbad62cb5